From ea3c504fae55c376192807c56d6fb005c3d7dda6 Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Tue, 14 Jan 2025 18:29:07 +0100
Subject: [PATCH 105/120] watchdog: don't autostart watchdog on STM32MP boards

The STM32MP2 boards have watchdog started by a previous boot
stage (e.g. bootrom or secure OS), WATCHDOG_AUTOSTART is useless.
Prefer to rely on DT properties "u-boot,autostart" or "u-boot,noautostart"
if needed.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git c9cb9d959de1c35ce76c4af3f42185dcbee5efd6 watchdog: don't autostart watchdog on STM32MP boards]
Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Change-Id: I41e5db56663c52570bd66c7105b41d737668a7e7
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/428871
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/watchdog/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/watchdog/Kconfig b/drivers/watchdog/Kconfig
index 1bb67f50352..510330be8a5 100644
--- a/drivers/watchdog/Kconfig
+++ b/drivers/watchdog/Kconfig
@@ -15,6 +15,7 @@ config WATCHDOG_AUTOSTART
 	bool "Automatically start watchdog timer"
 	depends on WDT
 	default n if ARCH_SUNXI
+	default n if ARCH_STM32MP
 	default y
 	help
 	  Automatically start watchdog timer and start servicing it during
-- 
2.47.2

